_posts/2020-09-22-data-explorer-kafka-connect.md

```diff
@@ -0,0 +1,413 @@
+
+### Azure Data Explorer
+
+You can setup an Azure Data Explorer cluster and database [using Azure Portal](https://docs.microsoft.com/azure/data-explorer/create-cluster-database-portal), [Azure CLI](https://docs.microsoft.com/azure/data-explorer/create-cluster-database-cli) or any of the client SDKs such as [Python](https://docs.microsoft.com/azure/data-explorer/create-cluster-database-python). Once that's done, create a table (named `Storms`) and respective mapping (named `Storms_CSV_Mapping`) using below queries:
+
+```kusto
+.create table Storms (StartTime: datetime, EndTime: datetime, EventId: int, State: string, EventType: string, Source: string)
+
+.create table Storms ingestion csv mapping 'Storms_CSV_Mapping' '[{"Name":"StartTime","datatype":"datetime","Ordinal":0}, {"Name":"EndTime","datatype":"datetime","Ordinal":1},{"Name":"EventId","datatype":"int","Ordinal":2},{"Name":"State","datatype":"string","Ordinal":3},{"Name":"EventType","datatype":"string","Ordinal":4},{"Name":"Source","datatype":"string","Ordinal":5}]'
+```
+
+### Azure Kubernetes Service (optional)
+
+I have used [Azure Kubernetes Service](https://docs.microsoft.com/azure/aks/) (AKS) but the instructions in this blog post should work for other options as well (e.g. with a local `minikube` cluster on your laptop). You can setup an AKS cluster using [Azure CLI](https://docs.microsoft.com/azure/aks/kubernetes-walkthrough), [Azure portal](https://docs.microsoft.com/azure/aks/kubernetes-walkthrough-portal) or [ARM template](https://docs.microsoft.com/azure/aks/kubernetes-walkthrough-rm-template) 
+
+## Base installation
+
+Start by installing the Strimzi Operator and use it to spin up a single-node Kafka Cluster on Kubernetes. Installing Strimzi using `Helm` is pretty easy:
+
+```shell
+helm repo add strimzi https://strimzi.io/charts/
+helm install strimzi-kafka strimzi/strimzi-kafka-operator
+```
+
+To confirm successful installation:
+
+```
+kubectl get pods -l=name=strimzi-cluster-operator
+
+//output:
+NAME                                        READY   STATUS    RESTARTS   AGE
+strimzi-cluster-operator-5c66f679d5-69rgk   1/1     Running   0          43s
+```
+
+To deploy a single-node kafka cluster (along with Zookeeper):
+
+```shell
+kubectl apply -f https://github.com/abhirockzz/kusto-kafka-connect-strimzi/raw/master/deploy/kafka.yaml
+
+//wait for the cluster to start
+kubectl get pod my-kafka-cluster-kafka-0 -w
+```
+
+## Kafka Connect cluster setup
+
+The Strimzi container images for Kafka Connect include two built-in file connectors - `FileStreamSourceConnector` and `FileStreamSinkConnector`. The Azure Data Explorer connector is installed on top of the [Strimzi Kafka Docker image](https://hub.docker.com/r/strimzi/kafka) by simply adding the connector plugin JAR to the plugin path as such:
```

Comment:
I guess you should `wget` the jar that you're copying?

Reply:
thanks. I have revamped this section to include more details